### ISSUE TITLE
[Bug Report] Different results from HuggingFace when using the GPT2 small example

### ISSUE THREAD

**Issue by nreHieW** [TransformerLensOrg/TransformerLens#685]
If you are submitting a bug report, please fill in the following details and use the tag [bug].

**Describe the bug**
The output from gpt2 small is different form the hf implementation. I searched for gpt2 implementation specifically and there doesn't seem to be anything 

**Code example**
Please try to provide a minimal example to reproduce the bug. Error messages and stack traces are also helpful.

```python
import transformer_lens
import torch
from transformers import AutoTokenizer, AutoModelForCausalLM

transformer_lens_model = transformer_lens.HookedTransformer.from_pretrained("gpt2-small").cuda()
transformer_lens_model.cfg.default_prepend_bos = False

logits, activations = transformer_lens_model.run_with_cache("Hello World")


tokenizer = AutoTokenizer.from_pretrained("gpt2")
model = AutoModelForCausalLM.from_pretrained("gpt2").cuda()
inputs = tokenizer("Hello World", return_tensors="pt").to("cuda")
outputs = model(**inputs, output_hidden_states=True)
torch.allclose(logits, outputs.logits)
```

**System Info**
Describe the characteristic of your environment:
This was run on the free tier of Google Colab 

**Additional context**

### Checklist

- [x] I have checked that there is no similar [issue](https://github.com/TransformerLensOrg/TransformerLens/issues) in the repo (**required**)

**Comment by degenfabian** [comment]
Hi!

Thanks for sharing the code you used! I notice that you are loading the model with HookedTransformer.from_pretrained().
There is nothing wrong with that, but when comparing logits you should always use HookedTransformer.from_pretrained_no_processing(), because by default TransformerLens centers the unembedding, which translates every logit by a fixed amount per token.

With loading in the model via from_pretrained_no_processing(), the logits turn out to be very well aligned:

```
=== Logits Comparison ===
TransformerLens:
Mean: tensor(-59.5233, device='mps:0', grad_fn=<MeanBackward0>)
Std: tensor(16.6638, device='mps:0', grad_fn=<StdBackward0>)
HuggingFace:
Mean: tensor(-59.5232, device='mps:0', grad_fn=<MeanBackward0>)
Std: tensor(16.6639, device='mps:0', grad_fn=<StdBackward0>)
Max diff: tensor(0.0003, device='mps:0', grad_fn=<MaxBackward1>)
```
